File: v1_backup/backend/src/routes/notifications.js
const express = require('express');
const router = express.Router();
const pool = require('../config/database');

router.get('/:userId', (req, res) => {
  try {
    const { unread_only = 'false', limit = 50 } = req.query;
    let query = `
      SELECT n.*, c.contract_no, c.title as contract_title
      FROM notifications n
      LEFT JOIN contracts c ON n.related_contract_id = c.id
      WHERE n.user_id = ?
    `;
    const params = [req.params.userId];
    if (unread_only === 'true') query += ' AND n.is_read = 0';
    query += ' ORDER BY n.created_at DESC LIMIT ?';
    params.push(parseInt(limit));

    const result = pool.query(query, params);
    const unreadCount = pool.query(
      'SELECT COUNT(*) as count FROM notifications WHERE user_id = ? AND is_read = 0',
      [req.params.userId]
    );

    res.json({
      notifications: result.rows,
      unread_count: parseInt(unreadCount.rows[0].count)
    });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

router.put('/:id/read', (req, res) => {
  try {
    pool.query('UPDATE notifications SET is_read = 1 WHERE id = ?', [req.params.id]);
    res.json({ success: true });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

router.put('/read-all/:userId', (req, res) => {
  try {
    pool.query('UPDATE notifications SET is_read = 1 WHERE user_id = ? AND is_read = 0', [req.params.userId]);
    res.json({ success: true });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;

File: v1_backup/backend/routers/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from database import get_db
from models.user import User
from models.notification import Notification
from middleware.auth import get_current_user

router = APIRouter(prefix="/api/notifications", tags=["通知"])


@router.get("/")
def list_notifications(
    unread_only: bool = False,
    db: Session = Depends(get_db),
    user: User = Depends(get_current_user),
):
    q = db.query(Notification).filter(Notification.user_id == user.id)
    if unread_only:
        q = q.filter(Notification.is_read == False)
    items = q.order_by(Notification.created_at.desc()).limit(50).all()

    unread_count = db.query(Notification).filter(
        Notification.user_id == user.id, Notification.is_read == False
    ).count()

    return {
        "notifications": [
            {
                "id": n.id, "type": n.type, "message": n.message,
                "contract_id": n.contract_id, "is_read": n.is_read,
                "created_at": n.created_at.isoformat() if n.created_at else None,
            }
            for n in items
        ],
        "unread_count": unread_count,
    }


@router.post("/{notification_id}/read")
def mark_read(notification_id: int, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    notif = db.get(Notification, notification_id)
    if notif and notif.user_id == user.id:
        notif.is_read = True
        db.commit()
    return {"success": True}


@router.post("/read-all")
def mark_all_read(db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    db.query(Notification).filter(
        Notification.user_id == user.id, Notification.is_read == False
    ).update({"is_read": True})
    db.commit()
    return {"success": True}
